src/ui/common/src/components/pages/function/id/index.tsx

```diff
@@ -0,0 +1,409 @@
+import { exportFunction } from '../../../../utils/operators';
+import { LoadingStatusEnum } from '../../../../utils/shared';
+import DetailsPageHeader from '../../components/DetailsPageHeader';
+import { LayoutProps } from '../../types';
+
+type FunctionDetailsPageProps = {
+  user: UserProfile;
+  Layout?: React.FC<LayoutProps>;
+  maxRenderSize?: number;
+};
+
+const listStyle = {
+  width: '100%',
+  maxWidth: 360,
+  bgcolor: 'background.paper',
+};
+
+// Checked with file size=313285391 and handles that smoothly once loaded. However, takes a while to load.
+const FunctionDetailsPage: React.FC<FunctionDetailsPageProps> = ({
+  user,
+  Layout = DefaultLayout,
+  maxRenderSize = 100000000,
+}) => {
+  const dispatch: AppDispatch = useDispatch();
+  const navigate = useNavigate();
+  const [inputsExpanded, setInputsExpanded] = useState<boolean>(true);
+  const [outputsExpanded, setOutputsExpanded] = useState<boolean>(true);
+  const [files, setFiles] = useState({
+    '': {
+      path: '',
+      language: 'plaintext',
+      content: '',
+    },
+  });
+
+  const artifactTypeToIconMapping = {
+    [ArtifactType.String]: stringArtifactNodeIcon,
+    [ArtifactType.Bool]: boolArtifactNodeIcon,
+    [ArtifactType.Numeric]: numericArtifactNodeIcon,
+    [ArtifactType.Dict]: dictArtifactNodeIcon,
+    // TODO: figure out if we should use other icon for tuple
+    [ArtifactType.Tuple]: dictArtifactNodeIcon,
+    [ArtifactType.Table]: tableArtifactNodeIcon,
+    [ArtifactType.Json]: jsonArtifactNodeIcon,
+    // TODO: figure out what to show for bytes.
+    [ArtifactType.Bytes]: dictArtifactNodeIcon,
+    [ArtifactType.Image]: imageArtifactNodeIcon,
+    // TODO: Figure out what to show for Picklable
+    [ArtifactType.Picklable]: dictArtifactNodeIcon,
+  };
+
+  const params = useParams();
+  const workflow = useSelector((state: RootState) => state.workflowReducer);
+  const operator = workflow.operatorResults[params.operatorId];
+  const inputs = workflow.selectedDag?.operators[params.operatorId]?.inputs;
+  const outputs = workflow.selectedDag?.operators[params.operatorId]?.outputs;
+  if (inputs && outputs) {
+    const operatorArtifacts = [...inputs, ...outputs];
+    // fetch output artifacts.
+    operatorArtifacts.map((artifactId) => {
+      if (!workflow.artifactResults[artifactId])
+        dispatch(
+          handleGetArtifactResults({
+            apiKey: user.apiKey,
+            workflowDagResultId: params.workflowDagResultId,
+            artifactId: artifactId,
+          })
+        );
+    });
+  }
+
+  useEffect(() => {
+    document.title = 'Operator Details | Aqueduct';
+    if (
+      workflow.selectedDag === undefined ||
+      (workflow.selectedDag && !(params.workflowId in workflow.selectedDag))
+    ) {
+      dispatch(
+        handleGetWorkflow({
+          apiKey: user.apiKey,
+          workflowId: params.workflowId,
+        })
+      );
+    }
+  }, []);
+
+  useEffect(() => {
```

Comment:
@likawind should we be using your new API route for getting the file results here?

Reply:
I think we should do all route migration separately. We can get the functional part merged first.

For operator file in particular, we don't have another new route to get files yet (we assume file could be large so it's not included in the new route's response), so for file it's good to use the endpoint underlying `handleGetOperatorResults` now.